.github/workflows/ruff.yml

```diff
@@ -0,0 +1,69 @@
+# For most projects, this workflow file will not need changing; you simply need
+# to commit it to your repository.
+#
+# You may wish to alter this file to disable certain Ruff checks in the
+# "Analyze" configuration block below.
+#
+# For more information see:
+#     https://nasa-ammos.github.io/slim/docs/guides/software-lifecycle/application-starter-kits/python-starter-kit/
+#
+# ******** NOTE ********
+# Ruff is a Python-based linter that works to evaluate Python code.
+#
+name: "Ruff"
+
+on:
+  push:
+    branches: [main, develop]
+  pull_request:
+    # The branches below must be a subset of the branches above
+    branches: [develop]
+
+jobs:
+  analyze:
+    name: Analyze
+    runs-on: ubuntu-latest
+    permissions:
+      actions: write
```

Comment:
Does it need write access? That seems like we're handing the VM a lot of power.

Reply:
It does not as it's merely analyzing the code. I'm assuming I should take away its permission to write during security-events as well?